.pre-commit-config.yaml

```diff
@@ -1,41 +1,50 @@
+ci:
+  autofix_prs: false
+
 repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.0.1
+    hooks:
+      - id: check-merge-conflict
+      - id: check-case-conflict
+      - id: check-ast
+      - id: debug-statements
+      - id: check-toml
+      - id: check-json
+
+  - repo: https://github.com/pre-commit/pygrep-hooks
+    rev: v1.9.0
+    hooks:
+      # - id: python-check-blanket-noqa # XXX: disallow blanket # noqa
+      - id: python-check-blanket-type-ignore
+      - id: python-check-mock-methods
+      # - id: python-use-type-annotations # XXX: disallow type comments
+
+  - repo: https://github.com/pycqa/isort
+    rev: 5.10.1
+    hooks:
+      - id: isort
+        types: [python] # FIXME: pyi support that doesn't conflcit with black
+
   - repo: https://github.com/psf/black
-    rev: 20.8b1
+    rev: 21.10b0
     hooks:
       - id: black
 
   - repo: https://gitlab.com/pycqa/flake8
-    rev: 3.8.4
+    rev: 4.0.1
     hooks:
       - id: flake8
-        additional_dependencies: [flake8-bugbear]
+        additional_dependencies:
+          # - flake8-docstring # XXX: lint docstrings
```

Comment:
These are a harder sell, but still are useful tools. I think the mypy project should be completed before either. Some of the non-PEP8 compliant names may be part of the exposed external API and therefore require at least a minor version change to be corrected.

Reply:
I'd honestly remove unused hooks for now, and add them as and when we enable them.